keras/src/layers/layer.py

```diff
@@ -1560,6 +1606,61 @@ def _open_name_scope(self):
             self._parent_path = current_path()
         return backend.name_scope(self.name, caller=self)
 
+    def rematerialized_call(self, layer_call, *args, **kwargs):
+        """Enable rematerialization dynamically for layer's call method.
+
+        Args:
+            layer_call: The original `call` method of a layer.
+
+        Returns:
+            Rematerialized layer's `call` method.
+        """
+        if any(isinstance(x, KerasTensor) for x in tree.flatten(args)):
+            # Bypass rematerialization when in symbolic mode.
+            return layer_call(*args, **kwargs)
```

Comment:
I think you don't need this if `rematerialize_call` is called from `call` only and not from `__call__`.